File: client/lib/monarch/model/projected_column.js
(function(Monarch) {

_.constructor("Monarch.Model.ProjectedColumn", {
  initialize: function(column, columnAlias) {
    this.column = column;
    this.columnAlias = columnAlias;
  },

  name: function() {
    return this.columnAlias || this.column.name;
  },

  eq: function(rightOperand) {
    return new Monarch.Model.Predicates.Eq(this, rightOperand);
  },

  asc: function() {
    return new Monarch.Model.SortSpecification(this, 'asc');
  },

  desc: function() {
    return new Monarch.Model.SortSpecification(this, 'desc');
  }
});

})(Monarch);

File: server/lib/monarch/model/projected_column.rb
module Model
  class ProjectedColumn
    attr_reader :column, :column_alias

    delegate :convert_value_for_storage, :to => :column

    def initialize(column, column_alias=nil)
      @column, @column_alias = column, column_alias
    end

    def name
      column_alias || column.name
    end

    def to_sql
      as_suffix = column_alias ? " as #{column_alias}" : ""
      "#{column.to_sql}#{as_suffix}"
    end
  end
end
